### DISCUSSION TITLE
Please make a helm chart out of the docker-compose.yaml

### DISCUSSION THREAD

**Discussion by EasternPA** [teslamate-org/teslamate#1071] [Ideas]
**What would you like to be added**: Helm chart

**Why is this needed**: Cloud hosting is getting simplified, and even helm is becoming easier to use at home

**Additional context**:

I'm using the 'kompose' utility in Linux to convert the docker-compose.yml to a helm chart. I'm working through the bumps in the road, but the current one that I'm running into is

`initdb: error: directory "/var/lib/postgresql/data" exists but is not empty
It contains a lost+found directory, perhaps due to it being a mount point.
Using a mount point directly as the data directory is not recommended.
Create a subdirectory under the mount point.`

Could you either `rm -rf lost+found` or make a new directory and stick the database in there to help get around this?

**Comment by EasternPA** [comment]
Please ignore prior comments. I started over with the teslamate helm chart and just ran teslamate/grafana from dockerhub in the same namespace. Anyone can hit me up if you want to talk about it.

**Reply by cpouthier** [reply]
Hello, I tried to "kompose convert" from the [docker-compose.yml ](https://docs.teslamate.org/docs/installation/docker/)file but unfortunately I ran into errors. How can I get rid of it?
 
**WARN Service "database" won't be created because 'ports' is not specified 
WARN Service "mosquitto" won't be created because 'ports' is not specified 
WARN Volume mount on the host "/home/ubuntu/teslamate/import" isn't supported - ignoring path on the host** 
INFO Kubernetes file "grafana-service.yaml" created 
INFO Kubernetes file "teslamate-service.yaml" created 
INFO Kubernetes file "database-deployment.yaml" created 
INFO Kubernetes file "teslamate-db-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "grafana-deployment.yaml" created 
INFO Kubernetes file "teslamate-grafana-data-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "mosquitto-deployment.yaml" created 
INFO Kubernetes file "mosquitto-conf-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "mosquitto-data-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "teslamate-deployment.yaml" created 
INFO Kubernetes file "teslamate-claim0-persistentvolumeclaim.yaml" created

**Comment by EasternPA** [comment]
I haven't run helm/k8s in years and forgot I opened this issue. The 'ports' errors tell me the kompose convert didn't pick up the ports declaration in the docker-compose.yml. The volume error means the docker-compose.yml is using an approach that k8s/helm doesn't support. You're probably better off just writing the helm yourself or writing a new docker-compose with the issues fixed.

Keep in mind that this whole utility is expected to stop working in a few weeks, so if you're using this to learn helm/k8s, you may wish to find another project with a longer time horizon. Sorry I couldn't be more help, I haven't used this method in years.

**Reply by cpouthier** [reply]
Thank you.

**Comment by brianmay** [comment]
I use to use and maintain helm charts for Teslamate. See:

* https://github.com/brianmay/charts/tree/main/charts/teslamate
* https://github.com/brianmay/charts/tree/main/charts/teslamate-grafana
* https://github.com/brianmay/charts/tree/main/charts/mosquitto (just a reference to an existing mosquitto chart)
* https://github.com/brianmay/charts

I didn't do postqresql in k8s. But I think there are charts for that also.

**Reply by cpouthier** [reply]
Intersting, thank you
